### ISSUE TITLE
Lab 6: Algorithms

### ISSUE THREAD

**Issue by lmock** [cs10/bjc-r#427]
Please comment below to leave us your lab reviews! Remember, please follow this format:

```
Your Name:

Overview of Lab Comments: i.e. Was this lab good or bad? Too fast? Too Slow? Does it need something new?

For specific issues: (you should copy this section as many times as necessary)
Page Link:
Type of Issue: i.e. (code) bug, grammar, typo, lack of clarity, etc
Severity: i.e. "prevents students from continuing", "minor inconvenience", etc
Comments: Description of the problem, what can be done to fix it, etc.
```

**Comment by judyworkman** [comment]
Judy Workman

Overview of Lab Comments: 
There is a lot of reading and background information in this lab.  Many students skip over the reading to just go straight to the coding.  I think the information needs to be condensed into bullet points or presented in some sort of way that is more concise as to help students understand better.  I also think it would be helpful to include some notes on runtime in this lab.  Nevertheless, the lab is overall good.  The interactive part led by the TA is very beneficial to students’ understanding of the material in addition to being very fun.

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=4
Type of Issue: typo
Severity: minor inconvenience
Comments: Above the block called “find the number (_) in the sorted list [_](memoized)” it says “This block will be very similar to our the second block you built, the sorted number finder. “ It should just say “the second block you built.”

**Comment by tcabs17** [comment]
Tori Cabot

Overview of Lab
The algorithms lab is well-paced and relatively straight forward, though still challenging. Perhaps it is too wordy in the beginning. I enjoyed the teacher-led activity when I was in CS10 and I think it is an engaging way to start off lab. Also, there should be an interactive multiple-choice quiz somewhere in this lab. The abundance of text is overwhelming and might lead to students skipping the reading, even though those paragraphs will help them. The last section of the lab is completely text and I would not be surprised if students skip it or only skim it.
(For Week September 8 - 12): There is also an issue in the order of the labs in general. Tuesday September 9th, the lab was about abstraction, Thursday the scheduled lab is on lists, and the first homework is due Friday. Lists are an important tool for completing HW1. I think in future semesters, the lists lab should be more than a day before the first homework is due.

Specific Issues
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=3
Type of Issue: Assignment clarity
Severity: Inconvenience
Comments: Perhaps the directions should say something about accounting for whether or not the number is actually in the random list when building the sorting blocks. That concept could trip up some students for a while.

**Comment by 333lisajin** [comment]
Lisa (Minqiao) Jin:

Overview of Lab Comments: This is a pretty decent lab. I believe the general information can really give an edge of what algorithm means and it is very easy to comprehend. I believe if more examples could be used if possible.

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4

Type of Issue: Too wordy. 

Severity: Students may lose patience on this page. 
Comments: Cut this page into two pages and make it more clearly.

**Comment by tinatu** [comment]
Tina Tu 

Overview of Lab Comments: Overall, this lab was very informative and well-put together.  I felt that it was easy to follow and I particularly liked how algorithms are compared to things that students can relate more easily to.  I felt the lab had decent examples, both in quantity and in quality.  Besides minor issues, this lab was great!

For specific issues: 
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1
Type of Issue: Formatting
Severity: Mild
Comments: The word "software" is at the very of left underneath the picture when instead, I feel that it should be lined up with the paragraph to which it belongs.

For specific issues:
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: Typo
Severity: Mild
Comments: In the "Searching Sorted Data" section, there is a typo in the box listing how things should generally work.  In number 2, it says "if it's is..." in the second line of the question instead of "if it is" or "if it's."

**Comment by LithiumH** [comment]
Your Name: Haojun Li

Overview of Lab Comments: Lab is simple concepts expressed in an interactive way. Although concepts of binary search and memoization might be hard, the lab works really well for students who are unfamiliar with these concepts. A bit too fast in the searching lab. Maybe split it up into two pages will help

For specific issues: (you should copy this section as many times as necessary)
Page Link: http://128.32.42.199/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of Issue: Coding
Severity: Mislead students
Comments: The starter file for memoization has a "wrap" inside the function. This will confuse student whether or not to use wrap. If used, the function could run really fast and thus mislead students that memoization is much more effective than it should be.

For specific issues: 
Page Link: http://128.32.42.199/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of Issue: Unclear message
Severity: Misdirect students
Comments: The starter file does not give a good example of memoization. Maybe consider changing the starter file a little bit to include examples like searching one element for 20 times rather than picking a random number every time.

**Comment by Laralinmcc** [comment]
Lara McConnaughey

Overview of Lab Comments: The lab has a lot of examples and activities which are very helpful when learning about algorithms. I think that maybe in the section where the student completes the find the number block they should also be asked to change the list length to show the how the list size will affect the run time. The lab does have a lot of reading and the last page is just text which is unappealing to students. Maybe add some more pictures. Overall the lab is well-paced, very informative and has some great examples.

For specific issues: 
Page Link: http://snap.berkeley.edu/snapsource/snap.html#open:http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/../../../prog/algorithms/algorithms-framework.xml
Type of Issue:  Confuses Students
Severity: i.e. minor inconvenience
Comments: The memorized block has a warp inside it which hasn't been explained to students yet.

**Comment by amrutayel** [comment]
Your Name: Amruta Yelamanchili

Overview of Lab Comments: The concepts of the lab are easy enough, but actually implementing the block for finding a number in a sorted list can be pretty brutal. Some students may easily be able to finish within a lab period, but many others will not. The first few pages have a lot of text and background that students will skip over, so the most important parts of those pages should be presented more concisely. 

For specific issues: 
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=4
Type of Issue: code/clarity
Severity: minor
Comments: The memoized find block has a warp included in it. Students haven’t learned about this yet, so it shouldn’t be used, or it should be used in all of the blocks.

Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/too-many-blocks.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=5
Type of Issue: clarity
Severity: minor
Comments: Not sure if this page wants students to actually attempt making just one find block, or if they should just think about how/why it would be done.

**Comment by numfah23** [comment]
Your Name: Natnicha Vanitchanant

Overview of Lab Comments: The lab was pretty short, but this was because of the non-computer activity with the numbered cups at the beginning. For the snap framework provided, having the different sort algorithms in different sprites was very helpful and made everything very well-organized. However, the given code used for testing might seem a bit complicated for the students to understand. Also, the tests take forever to run (not sure if this is an issue only on my laptop or on all computers because of snap’s speed). I would suggest letting the students practice writing their own test with fewer numbers in the list first to make sure that the code is working correctly. After all 3 of the sorts are implemented correctly, have them test the difference in time with the given tests.

For specific issues:
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1
Type of Issue: Typo
Severity: Minor inconvenience
Comments: “add an new explanation or circumstance to another student's contribution explaining when it might occur.” <- should be “add a new…”

For specific issues:
Page Link:https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of Issue: i.e. Typo
Severity: Minor inconvenience
Comments: This block will be very similar to our the second block you built, the sorted number finder. <- remove “our”.

**Comment by chrisk867** [comment]
Name: Chris Kilian

Overview of lab contents: I think that this is a really well crafted lab. The lab does have a lot of text, but the high amount of text is needed to explain what is going on. Binary search can be a difficult concept for people to understand, and even though the lab appears to be very wordy, I think that it is necessary. The binary search algorithm to find the element of the list in O(log n) is a pretty hard program to write, especially for bard new programmers. I remember struggling on the "find x in sorted list" function back when I took CS10. I think the combination of a lot of text for the students who are really interested and the bullet point summary on the bottom for the kids who want to just get to coding works very well.

For Specific Issues:
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&noassignment=true&noreading=true&novideo=true&course=cs10_fa14.html&step=4
Type of issue:  emphasis
Severity: minor / medium
Comments: I feel like we should give a little bit more on memoiziation. I remember reading this slide in CS10 and not thinking much of memoization. It wasn't until I took CS61B that I really got an appreciation of memoiziation. Memoization makes things extremely fast and is an amazing tool to have. It can help do crazy things like solve fib(2000) in a few seconds. I think we should make sure the students know just how much faster memoization can make a program.

**Comment by rxia752** [comment]
Rowena Xia

Overview: This lab is a good length and I like the inclusion of the TA-led hands-on activity, which is helpful for the understanding of the information.  There's a lot of text, but I feel like that is very necessary for helping the students understand in this section.  I do think that the last page needs to be changed to add some pictures or something because a block of just text is unappealing and I know that a lot of students aren't even going to read it.  And besides pictures, just some highlighting of important parts of the text would be helpful and keeping the attention of students.  

For Specific Issues: 
https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1
Type of issue: typo
Severity: minor 
Comments: In the last sentence, it should be "a new explanation" not "an new explanation."

**Comment by nish2020** [comment]
Name: Nishchala Singhal
Lab Overview: Overall, this is a good lab that introduces students to algorithms in an informative as well as creative manner. I like the analogies between an algorithm and recipe- they help the student further understand and familiarize themselves with the new, sometimes challenging concept. Also, the TA-led class activity makes the lab more interactive and helps to engage the students. Lastly, the lesser number of Snap! exercises is a good idea, as each problem is challenging and can take a while for students to solve. 

Specific Issues:
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/too-many-blocks.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=5
Type of Issue: Clarity
Severity: Minor
Comments: This is a minor issue, but since the last page just consists of paragraphs, students may tend to skip it. Maybe adding in some Snap! code examples, pictures, etc. would help.

**Comment by ghost** [comment]
Name: Anjli Shah
Lab Overview: Overall I think the lab is well done. Because a lot of the first problems is something discussed and then attempted, I think there might be room for another problem. Also the concept of algorithms is not reiterated after the first example. Other than that and wordiness, the lab does a good job of applying the material. 

Specific Issues:
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1 
Type of Issue: Students will not read
Severity: Minor reduction of level of understanding
Comments: This page is very wordy. I myself just scanned, so I don’t think student will read the whole thing but it’s important they think about the questions. Maybe put the questions as bullet points?

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4 
Type of Issue: Wording
Severity: Hard to understand/complete problem
Comments: Where it says “We’ll use a list to keep track of…the index of each element in the list will be the number we’re looking for,” is slightly confusing. Students might not understand that the index is equivalent to the number we are looking for because it is a sorted list.

**Comment by finnhowell** [comment]
Name: Fionnuala Howell
Lab Overview: This lab is useful to learn about algorithms, run time, and memoization. The lab included a lot of text, which students may just skip in order to get to the problems. It might be helpful at the beginning of the lab to emphasize the importance of reading through the labs, as it can get confusing about what the problems are. I thought that the information provided in the lab was more useful than the specific editing of sorted blocks. The memoization block in particular may be a challenging problem for students. 

Specific Issues:
Page link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&noassignment=true&noreading=true&novideo=true&course=cs10_fa14.html&step=4
Type of issue: Clarity
Severity: Medium
Comments: I found this problem to be confusing in that I did not know what they were asking of the block. 

Page link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=3
Severity: Minor
Comments: It could be confusing for students if they did not realize that the number may not be in the list, as it is generating random numbers from 1 - 400 but the length of the list is 200. Must take into account the number not being in the list. Or could change the code inside the startup file to just pick a random item in the list.

**Comment by mfujimoto** [comment]
Your Name: Mason Fujimoto

Overview of Lab Comments: I recall that this was a very concise and eye opening lab when I took CS10. I think the group activity is a great illustration of the mechanisms the students are learning. However, I think the TAs and LAs need to be enthusiastic in the group activity for it to be effective.

For specific issues: 
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=1
Type of Issue: I foresee most students skipping the group exercise on the bottom of the page.
Severity: Not severe.
Comments: Sadly, I think most students skip these kinds of excercises unless they know or have gotten to someone next to them. This may not justify changing the lab itself: just an observation and something to think about.

**Comment by carenthomas** [comment]
Name: Caren Thomas

Overview of Lab Comments: I liked that this lab was very straightforward - the parallel structure of all the tasks really helped eliminate confusion on what was expected for each activity, but clarity could be improved by condensing the amount of physical text for this lab. The first two pages could easily be combined into one page, and the "Searching Unsorted Data" and "Searching Sorted Data" sections should really be separated into two. 

Page Link:https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: Code takes too long to run
Severity: Inconvenience for students
Comments: It might just be my laptop, but testing the block for the unsorted data set was painfully slow (5+ minutes). I know an important point of the lab is to demonstrate the slow run time, but I feel like the same point could be made without slowly driving students toward insanity.. 

Page Link:https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: Overwhelming amount of text
Severity: Very minor inconvenience
Comments: This page is especially dense and can be improved by separating the two activities onto two different pages.

**Comment by joelterry** [comment]
Name: Joel Terry

Overview of Lab Comments: The trade-offs discussion activity is a bit wordy, it can easily be cut down to a sentence or two. I love the cup search activity, it's a great way to help students visualize the process of creating an efficient algorithm.

Specific Issues:
Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=4
Type of Issue: Typo
Severity: Insignificant
Comments: "This block will be very similar to OUR THE second block you built, the sorted number finder."

**Comment by william0cheung** [comment]
William Cheung

Overview of Lab Comments: Algorithms are hard to grasp at first, but the cup demonstration is definitely really helpful. Not a long lab, but the sorting will take up most of the time, which is fine because it is an important concept to master. Necessary and great lab! 

Specific issues: 
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=1
Type of Issue: lack of clarity
Severity: minor inconvenience
Comments: The last paragraph can be broken up so it's more clear. A suggestion would be making a new paragraph for (a), (b), and (c), since these are the instructions for the students to try. Also students may or may not want to just discuss these things with a partner unless they know each other quite well; maybe suggest actually coding something together instead of just talking without a tangible answer out there.

Specific issues:
Page link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=3
Type of issue: density of material
Severity: minor
Comments: I suggest making the "searching sorted data" its own slide; this is because there is a lot of info to take in and learn. If it is right after an already hard unsorted data, students may become discouraged. This is more of a psychological issue; if they accomplish the unsorted, they have the satisfaction of clicking the "next" button. :)

**Comment by erikdahlquist** [comment]
Erik Dahlquist

This lab does a nice job introducing a difficult concept and provides a welcome change of pace from the Snap exercise-grinding of previous labs, even though cup sorting isn't particularly adrenaline inducing. I think the large amount of text in the lab is permissible because there are few slides, activities to break up the reading, and the slides are relatively well organised. However, while I think the length is manageable, I do think the algorithm exercises may prove frustrating and overly time-consuming (especially the sorted and memoized blocks).
Praise for the sorting activities though. A welcome diversion.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=1
Type of Issue: typo
Severity: i.e. Shan't be noticed. Minimal
Comments: In the last sentence, "add an new explanation" should say "add a new explanation."

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: lack of clarity
Severity: Moderate
Comments: You may want to point out on the slides that in "framework provided" they should flip though the separate sprites, as each has some different code ready to go for each prompt.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: i.e. lack of clarity
Severity: moderate
Comments: You may want to reclarify that, just as with the unsorted block, the sorted block should return "0 " if the number is not in the list. Or I think it will not past the code skeleton's tests.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of issue: organisation
Severity: Minimal
Comments: Why put memoization on its own slide but put sorted and unsorted together? This could be left as-is to emphasize the importance of memoization, or the slides could be separated/re-organized.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of issue: lack of clarity
Severity: moderate
Comments: I think the students will likely not properly grasp the implementation or value of memoization from this slide. They don't really need to fully understand memoization (and I think it's also covered in lecture), so that's probably fine, but perhaps they should be walked through memoization a bit more.

**Comment by jeffz6** [comment]
Jeffrey Zhang

Overview of Lab Comments: I didn't like the lab very much.  The pacing felt off, with multiple pages in a row without coding, followed by some coding that students had a really hard time understanding.  The testing framework was nice, but definitely needed to run a lot fewer tests by default, as students definitely did not have time to sit around for several minutes every time they had a bug.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: clarity
Severity: minor
Comments: It's not stated anywhere that the framework for sorted list search is on a different sprite

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: time
Severity: moderate
Comments: the default numbers of test runs and list elements are excessive and will take forever.  they should be a lot smaller

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=3
Type of Issue: difficulty
Severity: moderate
Comments: this question was noticeably harder than everything else and students struggled a lot with it.  It could use more explanation

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of Issue: bug
Severity: minor
Comments: the given code sets everything in memory to 0, not -1

**Comment by valenzuelav94** [comment]
Vanessa Valenzuela 

Overview of Lab Comments: I think it's definitely possible to shorten the length of the information given in the beginning. It's all important information, but what's most significant is being able to code this. I thought the algorithms lab was slightly more complex than most of the previous labs. When I took this class I enjoyed the cup activity, maybe keeping that on it's own than throwing all that information to students would be best. (just a suggestion by the way!) 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/algorithms-in-snap.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=3
Type of Issue: lack of clarity, could be more concise 
Severity: minor 
Comments: the given information could be condensed, not everything that is given is necessary. also, where can "sort" be found? giving that information could be efficient for students. 

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=4
Type of Issue: lack of clarity 
Severity: moderate 
Comments: instead of having students try to figure this through a slide, it would be best if it were discussed in either a section or during lab; it's a hard concept to wrap around your head, especially if this is your first time programming. it's important to realize that this requires a different way of thinking and not everyone will be able to understand this through a slide, (maybe try different methods of teaching this information).

**Comment by strollinghome** [comment]
Carlos Flores

Overview:
The lab was ok up until memoization. A lot of students go very confused by it and I'm not sure if its because they didn't read the entire page or because it unclear. In my opinion it seems clear once you read the entire thing but at first glance I can see how it can be intimidating for students.

**Comment by adambrown13** [comment]
Name: Adam Brown

Overview of Lab Comments: It was full of really good, integral information. However, I think the front end could have a lot more practice problems. It's techniques and facts that students will have to use throughout the course, so they should be more heavily tested on it, in order to make sure that they truly understand the concepts.

For specific issues:
Page Link: http://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/trash-or-treasure.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=1
Type of Issue: Group Project realism
Severity: Unless they're already in groups, I don't think that they're likely to get in groups explicitly for this exercise

**Comment by melvinhernandez** [comment]
Melvin Hernandez

Overview of Lab Comments: The lab did a great job at simplifying what an algorithm is. One concern that I have is that at the beginning there was a lot of conceptional information, like the group exercise, that students may not think is useful. Presenting the first couple of pages in a more "important - look here!" way may draw their attention as opposed to just presenting them with a ton of text for them to absorb. Overall, it was great although the binary and memoization ideas may be a bit difficult to understand at first.

Page Link: https://inst.eecs.berkeley.edu/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassignment=true&course=cs10_fa14.html&step=4
Type of Issue: Confusion
Comment: With the code given, one of the blocks has the warp block inside and it might confuse a student since it was not explained.

**Comment by MicBrain** [comment]
Rafayel Mkrtchyan

Overview: Overall the lab was pretty well written. Concepts were mostly comprehensive. Binary search was well explained and most of the students got it immediately. The hardest part of the lab, I believe was memoization, because a lot of students were confused about the general logic of memoization and how it actually works.

Page Link: http://128.32.42.199/~cs10/labs/cur/programming/algorithms/improving-our-number-finder.html?topic=berkeley_bjc%2Fareas%2Falgorithms.topic&novideo=true&noreading=true&noassingment=true&course=cs10_fa14.html&step=4
Type of issue. Clarity.
Comments: I think that the idea is just a little bit complicated and too broad a first example,
